hibernate-core/src/main/java/org/hibernate/persister/entity/JoinedSubclassEntityPersister.java

```diff
@@ -1139,7 +1139,7 @@ public String getPropertyTableName(String propertyName) {
 		return constraintOrderedTableNames;
 	}
 
-	public String[][] getContraintOrderedTableKeyColumnClosure() {
+	public String[][] getConstraintOrderedTableKeyColumnClosure() {
```

Comment:
Changing these public contracts on the other hand might be problematic as that is not backwards compatible. WDYT @sebersole and @dreab8? Should we just do this? +1 from my side

Reply:
+1 for me, probably we can deprecate them in 5.5, adding the new method method that calls the old one

Reply:
I may be wrong but it seems that in 6 we are not using this method anymore

Reply:
Probably.  I added that originally to support bulk manipulation queries, which are handled very differently now.

Still the point here is valid - this is a "public" API and we need to be careful changing those.  I vote we simply deprecate it

Reply:
I guess I have some knowledge gap here. We don't have freedom to change public method name in wip/v6?

Reply:
Yes and no. We try to do as few changes as possible to avoid unnecessary breakage. We might be able to remove some of the methods in the EntityPersister hierarchy so we might as well just rename some of them to fix grammar issues. I just wanted to bring it up so everyone can express their opinion on this topic.